### PR TITLE
HIRES wavelengths

### PR DESCRIPTION
**Pull request by profxj**
Testing for https://github.com/pypeit/PypeIt/pull/1628

### CODE REVIEW

**Comment by profxj** [comment]
this is a reminder to myself to remove the `setups` list
in `test_setups.py` before merging

File: vet_tests/test_wavelengths.py [hunk 1]
```diff
@@ -78,3 +87,51 @@ def test_mdm_modspec(redux_out):
...
+    rdx_dir = os.path.join(redux_out,
+                             'shane_kast_red',
+                             setup)
+    # Artificially make the slit bad
+    slit_file = os.path.join(rdx_dir,
+                             'Calibrations',
+                             'Slits_A_0_DET01.fits.gz')
+    slits = SlitTraceSet.from_file(slit_file)
+    slits.mask[0] = slits.bitmask.turn_on(slits.mask[0], 'BADWVCALIB')
+    slits.to_file(slit_file, overwrite=True)
```
Comment (kbwestfall) [inline vet_tests/test_wavelengths.py:108]: I'm worried about overwriting the file.  Can we instead:
 - copy the original file
 - artificially make the slit bad in the original file
 - perform the test
 - delete the original (altered file) and move the copy back to the original file name

**Review by kbwestfall — CHANGES_REQUESTED** [review]
I'm "requesting changes" so that we can discuss the following three points:

 - I'm a little worried about how you're importing stuff from local files by adding the local directory to the path.  I give you a way around this that feels cleaner, but let me know what you think.
 - I'd prefer that we not overwrite the original slits file in your test.
 - I'm wondering if there's a better/cleaner directory structure for the pypeit files that are used for the `reduce` tests vs. those that are needed for more bespoke vet/unit tests.

File: pypeit_files/shane_kast_red_redoslit_600_5000_d46.pypeit [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Auto-generated PypeIt file
```
[the remark this answers is not in the document]
Comment (profxj) [reply pypeit_files/shane_kast_red_redoslit_600_5000_d46.pypeit:1]: that is a sensible suggestion.  I've moved that file into `vet_tests/files`

File: test_scripts/pypeit_tests.py [hunk 3]
```diff
@@ -623,11 +623,21 @@ def template_coadd2d_file(dev_path, instr, setup):
...
     Use template pypeit file to write the pypeit file relevant to the
     exising directory structure.
 
+    Args:
+        pyp_file (str): Path to the pypeit file to be corrected.
+        dev_path (str): Path to the development directory.
+        raw_data (str): Full path to the raw data for this PypeIt file
+        instr (str): Instrument name.
+        setup (str): Setup name.
+        rdxdir (str): Path to the reduction directory.
```
[the remark this answers is not in the document]
Comment (profxj) [reply test_scripts/pypeit_tests.py:639]: done

File: test_scripts/setups.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
+""" Test setups """
+
+
+# This dict specifies all the instruments and setups that are supported by the dev suite.
+#  The keys are the instruments and the values are a list of the supported setups.
+_all = {
```
[the remark this answers is not in the document]
Comment (profxj) [reply test_scripts/setups.py:6]: done

File: test_scripts/setups.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+        #'HS1700+6416_H45aH_RED_B2_ECH_0.00_XD_-0.00_1x2',  # This one has bad HDUs
+        #'J0100+2802_H204Hr_RED_C1_ECH_0.75_XD_1.69_1x2',
+        'J0100+2802_H204Hr_RED_C1_ECH_-0.82_XD_1.62_1x2',
+        'J0100+2802_H237Hr_RED_C1_ECH_0.88_XD_1.46_1x2',
+        'J0100+2802_H237Hr_RED_C1_ECH_-0.91_XD_1.46_1x2',
+        'J0100+2802_N255Hr_RED_C2_ECH_0.74_XD_1.39_1x3',
+        'J0306+1853_U074_RED_C2_ECH_0.72_XD_1.42_1x3',
+        'J0306+1853_U074_RED_C2_ECH_-0.86_XD_1.31_1x3',
+        'J1723+2243_W241_RED_C5_ECH_0.08_XD_0.90_2x2',
+        'J1723+2243_W241_RED_C5_ECH_-0.15_XD_0.90_2x2',
```
[the remark this answers is not in the document]
Comment (profxj) [reply test_scripts/setups.py:32]: I fear not  :)  

Could remove `ECH` but what's 3 characters..

File: vet_tests/test_echelles.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+import sys
+import glob
+import numpy as np
+
+from pypeit import spec2dobj
+
+# 
+sys.path.append(os.path.join( os.path.abspath(
+        os.environ["PYPEIT_DEV"]),"test_scripts"))
+import setups
```
[the remark this answers is not in the document]
Comment (profxj) [reply vet_tests/test_echelles.py:12]: ok, I'm doing this.  Wasn't a big fan of the absolute paths either.

This method is now in `pypeit.io`

File: vet_tests/test_echelles.py [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    """ Confirm that all of the orders processed fine for each setup"""
+    instr = 'gemini_gnirs_echelle'
+
+    chk_orders(instr, redux_out)
+
+def test_magellan_fire_orders(redux_out):
+    """ Confirm that all of the orders processed fine for each setup"""
+    instr = 'magellan_fire'
+
+    chk_orders(instr, redux_out)
```
[the remark this answers is not in the document]
Comment (profxj) [reply vet_tests/test_echelles.py:63]: fixed (at least this one)

File: vet_tests/test_wavelengths.py [hunk 8]
```diff
@@ -1,11 +1,20 @@
...
+from pypeit.slittrace import SlitTraceSet
+from pypeit.scripts.run_pypeit import RunPypeIt
 
 import pytest
 
+# 
+sys.path.append(os.path.join(
+    os.path.abspath(
+        os.environ["PYPEIT_DEV"]),"test_scripts"))
+import pypeit_tests
```
[the remark this answers is not in the document]
Comment (profxj) [reply vet_tests/test_wavelengths.py]: replaced

File: vet_tests/test_wavelengths.py (hunk 1 above)
Comment (profxj) [reply vet_tests/test_wavelengths.py:114]: done

**Review by kbwestfall — APPROVED** [review]
Looks great!

**Pull request merged**